scipp-imaging-2D.ipynb

```diff
@@ -224,218 +162,516 @@
-    "\n",
-    "\n",
-    "stitched.labels[\"detector_mapping\"] = make_detector_groups(324, 324, nx_target, ny_target)\n",
-    "stitched.labels[\"detector_mapping\"]"
+    "# Normalize by open beam\n",
+    "normalized = stitched / reference\n",
+    "\n",
+    "# Replace special values nan and inf\n",
+    "replacement=sc.Variable(value=0.0, variance=0.0)\n",
+    "kwargs = {\"nan\" : replacement, \"posinf\" : replacement, \"neginf\" : replacement}\n",
+    "for k in normalized.keys():\n",
+    "    sc.nan_to_num(normalized[k].data, out=normalized[k].data, **kwargs)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "stitched[\"normalized\"] = stitched[\"sample\"] / stitched[\"reference\"]\n",
-    "\n",
-    "zero_special_values(stitched[\"normalized\"].values)"
+    "if output_tiff_stack:\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample\", x_len=324, y_len=324,\n",
+    "                              base_name=\"initial_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_initial\"))\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample_elastic\", x_len=324, y_len=324,\n",
+    "                              base_name=\"elastic_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_elastic\"))\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample_plastic\", x_len=324, y_len=324,\n",
+    "                          base_name=\"elastic_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_plastic\"))"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
-    "scrolled": true
+    "pycharm": {
+     "name": "#%%\n"
+    }
    },
    "outputs": [],
    "source": [
-    "dm1d = sc.groupby(stitched[\"normalized\"], \"detector_mapping\", Dim.Row)\n",
-    "stitched[\"normalized_grpd\"] = dm1d.sum(Dim.Spectrum)"
+    "# Load in IDF for positions of source/sample...etc\n",
+    "sc.compat.mantid.load_component_info(normalized, instrument_file)\n",
+    "wavelength = sc.neutron.convert(normalized, \"tof\", \"wavelength\")\n",
+    "\n",
+    "# Position data ...etc. is dropped on conversion but is later required for plotting\n",
+    "sc.compat.mantid.load_component_info(wavelength, instrument_file)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": true
-   },
    "outputs": [],
    "source": [
-    "# Plot the first group\n",
-    "sc.plot.plot(sc.sum(stitched[\"normalized_grpd\"], Dim.Row)) if do_plots else None"
-   ]
+    "# Precalculate rather than doing this every loop, this will be removed in the future\n",
+    "neighbour_spectra = operations.generate_neighbouring_spectra(num_spectra=num_spectra, bank_width=bank_width)\n",
+    "\n",
+    "def median_filter(dataset, variable_key):\n",
```

Comment:
You mentioned something in the call on why you had to do this pixel-by-pixel, can you elaborate? It just looks tedious, inefficient, and most importantly error prone.

Reply:
I agree, see above comment ^.
I reached a point where I just burnt out looking at this dreaded function and reverted it. 

Since this produces valid data in its current state I'd rather get this PR merged then come back to it. This way the users (one of which has < 1 month left) can use this whilst we rewrite this in parallel

Reply:
Just a comment on performance - the median filter step was very slow for me.